hierarchicalforecast/methods.py

```diff
@@ -460,11 +460,11 @@ def _get_PW_matrices(
         self,
         S: sparse.csr_matrix,
         y_hat: np.ndarray,
-        tags: Dict[str, np.ndarray],
+        tags: Optional[Dict[str, np.ndarray]] = None,
```

Comment:
https://github.com/Nixtla/hierarchicalforecast/blob/485da8d9333b3570f1fefaaf101a9966ca3afd15/hierarchicalforecast/methods.py#L650-L654

We check for that the structural matrix encodes a strict hierarchy at the beginning of ``fit_predict``, so this allows us to both avoid creating ``tags`` for each cut node's subgraph and checking for strictness each time, because these subgraphs will also be strictly hierarchical.

Reply:
Ah, thanks!